tests/Aspire.Confluent.Kafka.Tests/MetricsTests.cs

```diff
@@ -0,0 +1,345 @@
+                        "name": "rdkafka#producer-1",
+                        "replyq": 100,
+                        "msg_cnt": 100,
+                        "msg_size": 1638400,
+                        "tx": 5,
+                        "tx_bytes": 1638400,
+                        "txmsgs": 5,
+                        "txmsg_bytes": 1638400,
+                        "rx": 5,
+                        "rx_bytes": 1638400,
+                        "rxmsgs": 5,
+                        "rxmsg_bytes": 1638400
+                    }
+                    """;
+            Add(new TestVariationData()
+            {
+                UseKeyed = true,
+                Statistics = [s1]
+            });
+            Add(new TestVariationData()
+            {
+                UseKeyed = false,
+                Statistics = [s1]
+            });
+        }
+    }
+
+    public class Get_ExposesStatisticsAsCountersAndGauge_AggregateCountersByName_TestVariations : TheoryData<TestVariationData>
+    {
+        public Get_ExposesStatisticsAsCountersAndGauge_AggregateCountersByName_TestVariations()
+        {
+            string s1 = """
+                    {
+                        "client_id": "rdkafka",
+                        "type": "producer",
+                        "name": "rdkafka#producer-1",
+                        "replyq": 100,
+                        "msg_cnt": 100,
+                        "msg_size": 1638400,
+                        "tx": 5,
+                        "tx_bytes": 1638400,
+                        "txmsgs": 5,
+                        "txmsg_bytes": 1638400,
+                        "rx": 5,
+                        "rx_bytes": 1638400,
+                        "rxmsgs": 5,
+                        "rxmsg_bytes": 1638400
+                    }
+                    """;
+            string s2 = """
+                    {
+                        "client_id": "rdkafka",
+                        "type": "producer",
+                        "name": "rdkafka#producer-1",
+                        "replyq": 200,
+                        "msg_cnt": 200,
+                        "msg_size": 3276800,
+                        "tx": 10,
+                        "tx_bytes": 3276800,
+                        "txmsgs": 10,
+                        "txmsg_bytes": 3276800,
+                        "rx": 10,
+                        "rx_bytes": 3276800,
+                        "rxmsgs": 10,
+                        "rxmsg_bytes": 3276800
+                    }
+                    """;
+            Add(new TestVariationData()
+            {
+                UseKeyed = true,
+                Statistics = [s1, s2]
+            });
+            Add(new TestVariationData()
+            {
+                UseKeyed = false,
+                Statistics = [s1, s2]
+            });
+        }
+    }
+
+    public record TestVariationData
+    {
+        public bool UseKeyed { get; set; }
+        public List<string>? Statistics { get; set; }
```

Comment:
```suggestion
        public required List<string> Statistics { get; set; }
```

Then you don't need to `!` operator on where this is used.

Reply:
done